### PR TITLE
Correct decryption methods for AEAD

### PR DESCRIPTION
**Pull request by adzialocha**
The given decryption implementations for AEAD in `rust_crypto` for the cases `ChaCha20Poly1305` and `Aes256Gcm` did not use the regarding decryption but encryption methods of `aes-gcm`. I assume this bug was introduced by copy / paste, this PR should fix it :+1:

### CODE REVIEW

File: rust_crypto/src/lib.rs [hunk 1]
```diff
@@ -184,14 +184,14 @@ impl OpenMlsCrypto for RustCrypto {
...
-                aes.encrypt(nonce.into(), Payload { msg: ct_tag, aad })
+                aes.decrypt(nonce.into(), Payload { msg: ct_tag, aad })
                     .map(|r| r.as_slice().into())
                     .map_err(|_| CryptoError::AeadDecryptionError)
             }
             AeadType::ChaCha20Poly1305 => {
                 let aes = ChaCha20Poly1305::new_from_slice(key)
                     .map_err(|_| CryptoError::CryptoLibraryError)?;
-                aes.encrypt(nonce.into(), Payload { msg: ct_tag, aad })
+                aes.decrypt(nonce.into(), Payload { msg: ct_tag, aad })
```
Comment (franziskuskiefer) [inline rust_crypto/src/lib.rs:194]: The variable name should get changed to something like `chacha_poly`.
```suggestion
                chacha_poly.decrypt(nonce.into(), Payload { msg: ct_tag, aad })
```

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks!
Indeed, looks like a copy paste issue. Interesting that it didn't blow up.
One more thing to update though.

**Comment by raphaelrobert** [comment]
Nice catch! I would also have suspected that this would have been caught by tests earlier (at least for ChaCha20Poly1305, AESGCM256 is not used by MLS). This means we need better testing, I filed #588.

**Comment by adzialocha** [comment]
I've changed the variable name to `chacha_poly` @franziskuskiefer  :+1:

> Nice catch! I would also have suspected that this would have been caught by tests earlier.

Yes, I was surprised myself, it started troubleshooting by reading the `aes-gcm` code and specification since the "decrypted" plaintext was suddenly 32 bytes larger and I was sure I didn't understand something about AES :sweat_smile:

**Comment by franziskuskiefer** [comment]
Clippy and codecov choked for some reason, I'll merge manually.

**Pull request merged**